infiniband/datadog_checks/infiniband/check.py

```diff
@@ -0,0 +1,103 @@
+# (C) Datadog, Inc. 2025-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import glob
+import os
+from typing import Any  # noqa: F401
+
+from datadog_checks.base import AgentCheck  # noqa: F401
+
+from .metrics import IB_COUNTERS, RDMA_COUNTERS
+
+
+class InfinibandCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'infiniband'
+
+    def __init__(self, name, init_config, instances):
+        super(InfinibandCheck, self).__init__(name, init_config, instances)
+        self.tags = self.instance.get('tags', [])
+        self.base_path = self.instance.get('infiniband_path', '/sys/class/infiniband')
+
+        # Allow for additional counters to be collected if configured
+        self.additional_counters = set(self.instance.get('additional_counters', []))
+        self.additional_hw_counters = set(self.instance.get('additional_hw_counters', []))
+
+        # Allow for specific counters to be excluded if configured
+        self.exclude_counters = set(self.instance.get('exclude_counters', []))
+        self.exclude_hw_counters = set(self.instance.get('exclude_hw_counters', []))
+
+        # Allow for specific devices to be excluded if configured
+        self.exclude_devices = set(self.instance.get('exclude_devices', []))
+
+        # Configure how metrics should be collected
+        self.collection_type = self.instance.get('collection_type', 'gauge')
+        if self.collection_type not in ['gauge', 'monotonic_count', 'both']:
+            raise Exception("collection_type must be one of: 'gauge', 'monotonic_count', 'both'")
+
+        # Test to see if the path exist. In containerized environments it's customary to mount it to /host
+        if not os.path.exists(self.base_path):
+            alternative_path = os.path.join('/host', self.base_path.lstrip('/'))
```

Comment:
I like the workaround however I don't know how I feel that it won't be transparent to the user that they are running the check against the alternative path. Could we potentially either include a note in the conf.yaml instead, or log a line about this?

Reply:
Yeah that make sense. I was going to include a snippet in the doc because we don't automatically mount the /sys/class/infiniband directory by default in a kubernetes env so it's an extra step that is needed. I can clarify it a bit more in the conf.yaml and include a log line sure.

Reply:
Great, thanks!